### DISCUSSION TITLE
Invalidating Unsubscribed query cache data

### DISCUSSION THREAD

**Discussion by surajrane42** [reduxjs/redux-toolkit#3677] [Q&A]
I am new to RTK query & might be missing some basic understanding. Can someone help me with below issue

We have used RTK query in our project. There's a module where two list are rendered on two different tabs. Let's say List1 & List2. Both the lists have different endpoints for fetching the data. Both the endpoints are defined in same slice.
```
export const vmsApi = createApi({
  reducerPath: 'vmsApi',
  baseQuery: axiosBaseQuery(),
  tagTypes: [
    'list1',
    'list2',
  ],
  endpoints: (builder) => ({
    getList1: builder.query<any, Record<string, string>>({
      query: ({ organizationId, subscriptionId, vmId }) => ({
        url: `mock/vms_1.json`,
        method: 'GET',
      }),
      providesTags: ['list1'],
    }),
    getList2 builder.query<any, Record<string, string>>({
      query: ({ organizationId, subscriptionId, vmId }) => ({
        url: `mock/vms_2.json`,
        method: 'GET',
      }),
      providesTags: ['list2'],
    }),
  }),
});
```

There is a mutation action performed on records of `List1`, after the action is successful we need to invalidate both the lists (because the record gets move from list1 to list2 & some other record details are updated)

I have tried using the      ` dispatch(vmsApi.util.invalidateTags(['list1','list2']))` after the action. This invalidates the list1 correctly but list2 isn't invalidated. One thing to note here is I have checked in redux dev tools `getList2` query isn't a active subscription because its not currently rendered over the UI (when user navigates to the other tab only then list2 will be rendered)

I am not sure if RTK doesn't invalidates in-active subscriptions. If yes, Is there any way through which we can invalidate in-active subscriptions cache data?

**Comment by EskiMojo14** [comment]
cache entries with no subscriptions will instead be cleaned up if one of their tags is invalidated - this means that if any components then subscribe to that query it'll then fetch freshly as needed.
